### PR TITLE
Client-server fixes with deactivate/activate for an autonomous entity

### LINKED ISSUE #11282: ClientServer - asserts on host in Multiplayer::NetworkTransformComponentController::OnActivate
**Describe the bug**

Assert during host+loadlevel in a client-server hosted scenario.

```
System: Handler is already registered to an event, binding a handler to multiple events is unsupported
(System) - Handler is already registered to an event, binding a handler to multiple events is unsupported
System: ------------------------------------------------
(System) - ------------------------------------------------
System: C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (125) : AZ::EventHandler<AZ::EntityId,AZ::EntityId>::Connect
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (125) : AZ::EventHandler<AZ::EntityId,AZ::EntityId>::Connect
System: C:\git\o3de\Gems\Multiplayer\Code\Source\Components\NetworkTransformComponent.cpp (173) : Multiplayer::NetworkTransformComponentController::OnActivate
(System) - C:\git\o3de\Gems\Multiplayer\Code\Source\Components\NetworkTransformComponent.cpp (173) : Multiplayer::NetworkTransformComponentController::OnActivate
System: C:\git\o3de\Gems\Multiplayer\Code\Source\Components\NetBindComponent.cpp (342) : Multiplayer::NetBindComponent::EnablePlayerHostAutonomy
(System) - C:\git\o3de\Gems\Multiplayer\Code\Source\Components\NetBindComponent.cpp (342) : Multiplayer::NetBindComponent::EnablePlayerHostAutonomy
System: C:\git\o3de\Gems\Multiplayer\Code\Source\MultiplayerSystemComponent.cpp (1305) : Multiplayer::MultiplayerSystemComponent::EnableAutonomousControl
(System) - C:\git\o3de\Gems\Multiplayer\Code\Source\MultiplayerSystemComponent.cpp (1305) : Multiplayer::MultiplayerSystemComponent::EnableAutonomousControl
System: C:\git\o3de\Gems\Multiplayer\Code\Source\MultiplayerSystemComponent.cpp (1350) : Multiplayer::MultiplayerSystemComponent::OnRootSpawnableReady
(System) - C:\git\o3de\Gems\Multiplayer\Code\Source\MultiplayerSystemComponent.cpp (1350) : Multiplayer::MultiplayerSystemComponent::OnRootSpawnableReady
System: C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzFramework::RootSpawnableNotifications,AzFramework::RootSpawnableNotifications,0,1>::Dispatcher<AZ::EBus<AzFramework::RootSpawnableNotifications,A
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzFramework::RootSpawnableNotifications,AzFramework::RootSpawnableNotifications,0,1>::Dispatcher<AZ::EBus<AzFramework::RootSpawnableNotifications,A
System: C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Policies.h (267) : AZ::EBusQueuePolicy<1,AZ::EBus<AzFramework::RootSpawnableNotifications,AzFramework::RootSpawnableNotifications>,AZStd::recursive_mutex>::Execute
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Policies.h (267) : AZ::EBusQueuePolicy<1,AZ::EBus<AzFramework::RootSpawnableNotifications,AzFramework::RootSpawnableNotifications>,AZStd::recursive_mutex>::Execute
System: C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::Scri
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::Scri
System: C:\git\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1368) : AZ::ComponentApplication::Tick
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1368) : AZ::ComponentApplication::Tick
```

**Assets required**
o3de-multiplayersample

**Steps to reproduce**
1. Launch a game launcher with a multiplayer configured
2. host
3. loadlevel
4. See assert in the log

**Expected behavior**
No assert is hit during host+loadlevel

**Actual behavior**
Asserts with 
```
System: Trace::Assert
 C:\git\o3de\Code\Framework\AzCore\AzCore/EBus/Event.inl(124): (50292) 'void __cdecl AZ::EventHandler<class AZ::EntityId,class AZ::EntityId>::Connect(class AZ::Event<class AZ::EntityId,class AZ::EntityId> &)'
(System) - Trace::Assert
 C:\git\o3de\Code\Framework\AzCore\AzCore/EBus/Event.inl(124): (50292) 'void __cdecl AZ::EventHandler<class AZ::EntityId,class AZ::EntityId>::Connect(class AZ::Event<class AZ::EntityId,class AZ::EntityId> &)'
System: Handler is already registered to an event, binding a handler to multiple events is unsupported
(System) - Handler is already registered to an event, binding a handler to multiple events is unsupported
System: ------------------------------------------------
(System) - ------------------------------------------------
System: C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (125) : AZ::EventHandler<AZ::EntityId,AZ::EntityId>::Connect
(System) - C:\git\o3de\Code\Framework\AzCore\AzCore\EBus\Event.inl (125) : AZ::EventHandler<AZ::EntityId,AZ::EntityId>::Connect
System: C:\git\o3de\Gems\Multiplayer\Code\Source\Components\NetworkTransformComponent.cpp (173) : Multiplayer::NetworkTransformComponentController::OnActivate
```

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows, macOS, iOS, Android]

**Additional context**

```
    void NetworkTransformComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
    {
        if (AzFramework::TransformComponent* parentTransform = GetParent().GetTransformComponent())
        {
            parentTransform->BindTransformChangedEventHandler(m_transformChangedHandler);
            OnTransformChangedEvent(parentTransform->GetLocalTM(), parentTransform->GetWorldTM());

            parentTransform->BindParentChangedEventHandler(m_parentIdChangedHandler);
            OnParentIdChangedEvent(AZ::EntityId(), parentTransform->GetParentId());
        }
    }
```

The culprit is likely that the AZ::Events are not disconnected: m_transformChangedHandler and m_parentIdChangedHandler in NetworkTransformComponentController::OnDeactivate

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>

## What does this PR do?

Fixes #11282
TLDR: an assert is being hit during host+loadlevel in a client-server hosted scenario.

## How was this PR tested?

Tested wIth a multiplayer project in client-server mode

### CODE REVIEW

**Review by AMZN-puvvadar — APPROVED** [review]

**Review by allisaurus — APPROVED** [review]

**Review by kberg-amzn — APPROVED** [review]

File: Gems/Multiplayer/Code/Source/Components/NetworkTransformComponent.cpp [hunk 1]
```diff
@@ -176,7 +176,8 @@ namespace Multiplayer
 
     void NetworkTransformComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
-        ;
+        m_parentIdChangedHandler.Disconnect();
```
Comment (AMZN-Gene) [inline Gems/Multiplayer/Code/Source/Components/NetworkTransformComponent.cpp:179]: Activate/Deactivate symmetry restored. Thank you!

**Review by AMZN-Gene — APPROVED** [review]
